webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,149 @@
+import { STATUS } from 'foremanReact/constants';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { Bullseye, Progress, ProgressVariant, Button, Stack, StackItem } from '@patternfly/react-core';
+import { ChartDonut } from '@patternfly/react-charts';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { Redirect } from 'react-router-dom';
+import {
+  selectPublishContentViewsError, selectPublishContentViews,
+  selectPublishContentViewStatus,
+} from './ContentViewPublishSelectors';
+import { selectPublishTaskPoll, selectPublishTaskPollStatus } from '../Details/ContentViewDetailSelectors';
+import getContentViews, { publishContentView } from '../ContentViewsActions';
+import getContentViewDetails, { pollPublishTask, handleFinishedTask } from '../Details/ContentViewDetailActions';
+import Loading from '../../../components/Loading';
+import EmptyStateMessage from '../../../components/Table/EmptyStateMessage';
+import { cvVersionPublishKey } from '../ContentViewsConstants';
+
+
+const CVPublishFinish = ({
+  cvId,
+  checkedItems, setCheckedItems,
+  forcePromote, description, setDescription,
+  setIsOpen, versionCount,
+}) => {
+  const dispatch = useDispatch();
+  const [redirect, setRedirect] = useState(false);
+  const [saving, setSaving] = useState(true);
+  const [polling, setPolling] = useState(false);
+  const [taskErrored, setTaskErrored] = useState(false);
+  const [taskErrors, setTaskErrors] = useState(null);
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: checkedItems,
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setCheckedItems([]);
+  }, []);
```

Comment:
I don't like `[]` for the dependency array, as it breaks the rules.  Options are
1. Add `setSaving, dispatch, cvId, versionCount, description, checkedItems, forcePromote, setDescription, setCheckedItems` to the dependency array;
2. Create a ref outside of the effect: `const done = useRef(false);` Then put `if (done.current) return;` at the top of the useEffect.  And set `done.current` to true after `setCheckedItems([]);`. Then you can just not pass any dependency array at all.

Reply:
I thought this was the hook way of running the effect once on page's first render..That is what we need to do here in the final step of the wizard. Call publish and display progress..

Reply:
It's true that `[]` will cause the effect only to run once, but that technique really only works cleanly if you have no dependencies.  Once we start using the hooks eslint rules `[]` is not going to fly; we'll have to do something else.

Reply:
I see in foreman they use dispatch etc in the dependency array..Let me update that and verify that it still works..